### PR TITLE
Nest.JS + Next.JS integrated example

### PR DESCRIPTION
**Pull request by ephillipe**
This is a simple example that show the usage of Next.js with TypeScript and Nest.js as API backend.

### CODE REVIEW

File: examples/with-nestjs/tslint.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
```
Comment (timneutkens) [inline examples/with-nestjs/tslint.json]: I don't think tslint is needed for an example 👍

File: examples/with-nestjs/next.config.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const withTypescript = require('@zeit/next-typescript')
+const withCss = require('@zeit/next-css')
```
Comment (timneutkens) [inline examples/with-nestjs/next.config.js]: This is not required for nest.js + next.js, so I would remove it.

File: examples/with-nestjs/package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "with-typescript",
+  "version": "1.0.0",
+  "scripts": {
+    "dev": "node --inspect index.js",
+    "build": "next build",
+    "start": "NODE_ENV=production node index.js",
+    "type-check": "tsc"
+  },
+  "dependencies": {
```
Comment (timneutkens) [inline examples/with-nestjs/package.json]: There's a lot of dependencies here. Generally we recommend splitting your frontend and backend, as React SSR wouldn't impact the performance of the API in that case.

**Review by timneutkens — CHANGES_REQUESTED** [review]
The example seems to be showing a lot more than just simple `next.render()`, it parses json on requests and adds a full API endpoint. This is not really needed for just rendering Next.js.

File: examples/with-nestjs/package.json (hunk 3 above)
Comment (ephillipe) [reply examples/with-nestjs/package.json]: Hi, I cleanup some unused dependencies. 
I agree about recommendation of split frontend and backend, but sometimes we need consume services with CORS issues. Or simple make some transformation on the response of webservices. For that situations I think this solution.

Comment (timneutkens) [reply examples/with-nestjs/package.json]: In that case I think this should be a separate repository 🕵️

Comment (ephillipe) [reply examples/with-nestjs/package.json]: I think this cold be useful to Next.js users. The [pass-server-data](https://github.com/zeit/next.js/tree/canary/examples/pass-server-data) example do same thing using Express directly. But use Nest.js for these situations gives more flexibility to developers.

Comment (timneutkens) [reply examples/with-nestjs/package.json]: Ideally the example should only show rendering, as APIs should be external. I'm actually not a fan of pass-server-data and might remove it soon as it shows an anti-pattern.

**Comment by timneutkens** [comment]
I'm going to close this PR as per my earlier comment: https://github.com/zeit/next.js/pull/5886#discussion_r241886655

I'd be very happy to merge an implementation of just rendering using nest.js 🙏 

Thank you very much for your contribution! 🙌

**Pull request closed without merging**